bitrepository-audit-trail-service/src/main/java/org/bitrepository/audittrails/collector/AuditTrailCollector.java

```diff
@@ -70,7 +72,10 @@ public AuditTrailCollector(Settings settings, AuditTrailClient client, AuditTrai
 
         this.settings = settings;
         this.timer = new Timer(true);
-        long collectionInterval = settings.getReferenceSettings().getAuditTrailServiceSettings().getCollectAuditInterval();
+        javax.xml.datatype.Duration collectAuditInterval =
```

Comment:
What's the reason for using two different Duration types? (Just curious)

Reply:
For working with time in Java we want to use java.time including its `Duration` class. It allows nice things like `nextRun = Instant.now().plus(interval);`.
For specifying a duration/an interval in the XML settings I found it appropriate to use the `xs:duration` type, which gets translated into a `javax.xml.datatype.Duration` in Java.
Despite having the same name the two classes are not directly compatible since the XML duration includes years, months and days, which the java.time duration does not. I have nevertheless defined a conversion that generally isn’t precise (because years, months and even days don’t always have the same length), but I find it workable.